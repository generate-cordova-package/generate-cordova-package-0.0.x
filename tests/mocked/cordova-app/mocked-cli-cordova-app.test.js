/* global jest, it, expect */
/* (see <https://github.com/standard/standard/issues/905>) */

const snapshotDiff = require('snapshot-diff')

const startMain = require('../../../lib/start-main-cli')

const renderCordovaAppTree = require('../../../lib/cordova-app/render')

const renderCordovaAppProperties = require('../../render-cordova-app/render-cordova-app-properties.config')

const mockMainPromptResponse = {
  moduleType: 'cordova-app'
}

// render with these properties is covered by another test case:
const mockAppResponseProperties = renderCordovaAppProperties

const mockOutputLog = []

jest.mock('gitconfig', () => ({
  get: options => {
    mockOutputLog.push({ gitconfig: options })

    return Promise.resolve({
      user: { name: 'chris', email: 'chris@chris.com' }
    })
  }
}))

jest.mock('prompts', () => (questions, _) => {
  // with options ignored in this test case

  mockOutputLog.push({ prompts: { questions } })

  return Promise.resolve(
    questions[0].name === 'moduleType'
      ? mockMainPromptResponse
      : mockAppResponseProperties
  )
})

jest.mock('moment', () => () => ({
  // (starting to look forward):
  year: () => 2023
}))

jest.mock('fs-tree', () => tree => {
  mockOutputLog.push({ fsTree: tree })
})

it('start and run mocked CLI to create cordova-app', async () => {
  await startMain()

  // NOTE that the results of `lib/cordova-app/render.js` are
  // checked in another test case - no need to check here
  expect(
    snapshotDiff(
      // with rendered tree in an array to make snapshot-diff happy
      [renderCordovaAppTree(mockAppResponseProperties)],
      mockOutputLog,
      // cleaner snapshot diffs
      { contextLines: 1, stablePatchmarks: true }
    )
  ).toMatchSnapshot()
})
