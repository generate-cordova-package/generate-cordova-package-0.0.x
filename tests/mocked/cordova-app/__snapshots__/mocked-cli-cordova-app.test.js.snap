// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`start and run mocked CLI to create cordova-app 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ --- --- @@
    Object {
+     \\"prompts\\": Object {
+       \\"questions\\": Array [
+         Object {
+           \\"choices\\": Array [
+             Object {
+               \\"title\\": \\"Cordova app\\",
+               \\"value\\": \\"cordova-app\\",
+             },
+             Object {
+               \\"disabled\\": true,
+               \\"title\\": \\"Cordova plugin\\",
+               \\"value\\": \\"cordova-plugin\\",
+             },
+           ],
+           \\"default\\": 0,
+           \\"initial\\": 0,
+           \\"message\\": \\"Select module type\\",
+           \\"name\\": \\"moduleType\\",
+           \\"type\\": \\"select\\",
+         },
+       ],
+     },
+   },
+   Object {
+     \\"gitconfig\\": Object {
+       \\"location\\": \\"global\\",
+     },
+   },
+   Object {
+     \\"prompts\\": Object {
+       \\"questions\\": Array [
+         Object {
+           \\"default\\": \\"demo-app\\",
+           \\"initial\\": \\"demo-app\\",
+           \\"message\\": \\"Cordova app name\\",
+           \\"name\\": \\"name\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"com.demoapp\\",
+           \\"initial\\": \\"com.demoapp\\",
+           \\"message\\": \\"Cordova app id\\",
+           \\"name\\": \\"id\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"demo app\\",
+           \\"initial\\": \\"demo app\\",
+           \\"message\\": \\"Cordova app description\\",
+           \\"name\\": \\"description\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"0.0.1-dev\\",
+           \\"initial\\": \\"0.0.1-dev\\",
+           \\"message\\": \\"Initial app version\\",
+           \\"name\\": \\"version\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"MIT\\",
+           \\"initial\\": \\"MIT\\",
+           \\"message\\": \\"Cordova app license\\",
+           \\"name\\": \\"license\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"chris\\",
+           \\"initial\\": \\"chris\\",
+           \\"message\\": \\"author name\\",
+           \\"name\\": \\"author\\",
+           \\"type\\": \\"text\\",
+         },
+         Object {
+           \\"default\\": \\"chris@chris.com\\",
+           \\"initial\\": \\"chris@chris.com\\",
+           \\"message\\": \\"author email\\",
+           \\"name\\": \\"email\\",
+           \\"type\\": \\"text\\",
+         },
+       ],
+     },
+   },
+   Object {
+     \\"fsTree\\": Object {
        \\"test-app\\": Object {
@@ --- --- @@
  \\",
+         },
        },"
`;
